arch/x86/entry.S

```diff
@@ -208,6 +208,26 @@ ENTRY(syscall_handler_entry)
     SYSRET
 END_FUNC(syscall_handler_entry)
 
+ENTRY(sysenter_handler_entry)
+    push %_ASM_CX
+    push %_ASM_DX
```

Comment:
As mentioned in https://github.com/KernelTestFramework/ktf/pull/284#discussion_r1305451998, all callee-clobbered registers should be saved. `syscall_exit` also expects `flags` to be the top stack word, which SYSENTER doesn't provide -- it needs to be pushed manually.

Also, SYSENTER enters the kernel with user FLAGS, only IF and VM are forced to zero. So we might want to sanitize it (e.g. unset TF, AC) before executing further kernel code?

Reply:
The `flags` are pushed onto kernel stack by `enter_usermode()` when the task is created, so this aspect should be fine AFAICS.

Right, we have to sanitize the flags for example with the mask used for SYSCALL's MSR_FMASK.

Reply:
Please add `MASK_USER_FLAGS` after `syscall_from_usermode`